src/libraries/System.Private.CoreLib/src/System/Globalization/CalendarData.Windows.cs

```diff
@@ -0,0 +1,445 @@
+        private const uint CAL_SMONTHDAY = 0x00000038;
+        private const uint CAL_SSHORTDATE = 0x00000005;
+        private const uint CAL_SLONGDATE = 0x00000006;
+        private const uint CAL_SYEARMONTH = 0x0000002f;
+        private const uint CAL_SDAYNAME7 = 0x0000000d;
+        private const uint CAL_SABBREVDAYNAME7 = 0x00000014;
+        private const uint CAL_SMONTHNAME1 = 0x00000015;
+        private const uint CAL_SABBREVMONTHNAME1 = 0x00000022;
+        private const uint CAL_SSHORTESTDAYNAME7 = 0x00000037;
+        private const uint CAL_SERASTRING = 0x00000004;
+        private const uint CAL_SABBREVERASTRING = 0x00000039;
+
+        private const uint ENUM_ALL_CALENDARS = 0xffffffff;
+
+        private const uint LOCALE_ICALENDARTYPE = 0x00001009;
+        private const uint LOCALE_SSHORTDATE = 0x0000001F;
+        private const uint LOCALE_SLONGDATE = 0x00000020;
+        private const uint LOCALE_SYEARMONTH = 0x00001006;
+
+        private bool LoadCalendarDataFromSystemCore(string localeName, CalendarId calendarId)
+        {
+            Debug.Assert(!GlobalizationMode.Invariant);
+
+            if (GlobalizationMode.UseNls)
+            {
+                return NlsLoadCalendarDataFromSystem(localeName, calendarId);
+            }
+
+            // If running using ICU on Windows we should honor user overrides using NLS and the rest from ICU
+            bool result = IcuLoadCalendarDataFromSystem(localeName, calendarId);
+
+            if (result && bUseUserOverrides)
+            {
+                NormalizeCalendarId(ref calendarId, ref localeName);
+                result &= CallGetCalendarInfoEx(localeName, calendarId, CAL_ITWODIGITYEARMAX, out this.iTwoDigitYearMax);
+
+                // They want user overrides, see if the user calendar matches the input calendar
+                CalendarId userCalendar = (CalendarId)CultureData.GetLocaleInfoExInt(localeName, LOCALE_ICALENDARTYPE);
+
+                // If the calendars were the same, see if the locales were the same
+                if (userCalendar == calendarId)
+                {
+                    string? shortDateOverride = CultureData.ReescapeWin32String(CultureData.GetLocaleInfoEx(localeName, LOCALE_SSHORTDATE));
+                    string? longDateOverride = CultureData.ReescapeWin32String(CultureData.GetLocaleInfoEx(localeName, LOCALE_SLONGDATE));
+                    InsertOrSwapOverride(shortDateOverride, ref this.saShortDates);
+                    InsertOrSwapOverride(longDateOverride, ref this.saLongDates);
+                }
+            }
+
+            return result;
+        }
+
+        private void InsertOrSwapOverride(string? value, ref string[] destination)
+        {
+            if (value != null)
+            {
+                int index = -1;
+                for (int i = 0; i < destination.Length; i++)
+                {
+                    if (destination[i] == value)
+                    {
+                        index = i;
```

Comment:
>index = i; [](start = 24, length = 10)

nit: you can simplify this by just doing the swap here and then return from the method. this will get rid of the index variable and avoid more checking later and should make the code much simpler. 

just a suggestion  you don't have to do it.

Reply:
Thanks, fixed.